addons/misra.py

```diff
@@ -1461,6 +1461,79 @@ def reportErrorIfVariableIsNotConst(variable, stringLiteral):
                         if usedParameter.isString and parameterDefinition.nameToken:
                             reportErrorIfVariableIsNotConst(parameterDefinition.nameToken, usedParameter)
 
+    def misra_8_2(self, data, rawTokens):
+        def getFollowingRawTokens(rawTokens, token, count):
+            following =[]
+            for rawToken in rawTokens:
+                if (rawToken.file == token.file and
+                        rawToken.linenr == token.linenr and
+                        rawToken.column == token.column):
+                    for _ in range(count):
+                        rawToken = rawToken.next
+                        # Skip comments
+                        while rawTokens and (rawToken.str.startswith('/*') or rawToken.str.startswith('//')):
+                            rawToken = rawToken.next
+                        if rawToken is None:
+                            break
+                        following.append(rawToken)
+
+                    return following
+
+        # Check arguments in function declaration
+        for func in data.functions:
+
+            # Zero arguments should be in form ( void )
+            if (len(func.argument) == 0):
+                zeroCallTokens = getFollowingRawTokens(rawTokens, func.tokenDef, 3)
```

Comment:
I think it's a mistake to remove the "void". I wonder if we can just remove the `Tokenizer::simplifyParameterVoid`. Could you check if that would confuse the symboldatabase for some C/C++ code?

Reply:
I removed Tokenizer::simplifyParameterVoid in my latest commit and updated the tests accordingly. This made it possible to check this case without using the raw tokens.